### PR TITLE
one-click trial landing page

### PR DESCRIPTION
**Pull request by goneri**
Jira Issue: <https://issues.redhat.com/browse/AAP-26231>

## Description

Landing page for the 90 days Trial.

image

## Testing

- Set `ANSIBLE_AI_ENABLE_ONE_CLICK_TRIAL` True
- Connect to the service with OIDC user, the Org should not have any key set
- You should be able to accept the T&C and start a Trial period

No API permission changes should be triggered at this stage. Another PR will come for that.

## Production deployment

- [x] This code change is ready for production on its own
- [ ] This code change requires the following considerations before going to production:

### CODE REVIEW

**Comment by Ranelim** [comment]
It looks good to me for now.
Note: talking with @goneri, We will add a margin between the Terms and Conditions and the primary button on a later session.

File: ansible_ai_connect/users/templates/users/home.html [hunk 1]
```diff
@@ -118,6 +118,16 @@ <h1 class="pf-c-title pf-m-lg">{{ project_name }}</h1>
               {% if user.rh_user_is_org_admin or user.rh_user_has_seat %}
                 <p>Role: {% if user.rh_user_is_org_admin %}administrator{% endif %}{% if user.rh_user_is_org_admin and user.rh_user_has_seat %}, {% endif %}{% if user.rh_user_has_seat %}licensed user{% endif %}</p>
               {% endif %}
+
```
Comment (manstis) [inline ansible_ai_connect/users/templates/users/home.html:121]: Did you want to leave the commented block in this PR or remove it all together?

File: ansible_ai_connect/users/templates/users/trial.html [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+
+          {% if deployment_mode != "saas" %}
+          <!-- Not supported -->
+          {% elif has_active_plan %}
+          <div class="ls_message_body">
+            You have an active <i>{{ has_active_plan.plan.name }}</i> that will end the {{ has_active_plan.expired_at }}
+          </div>
+
+
+          <!--
```
Comment (manstis) [inline ansible_ai_connect/users/templates/users/trial.html:44]: Did you want to leave this here?

File: ansible_ai_connect/users/templates/users/trial.html [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+              -->
+
+              {% elif has_expired_plan %}
+              You have an expired ({{ has_active_plan.plan.name }}).
+              {% else  %}
+              <form action={% url 'trial' %} method="post">{% csrf_token %}
+
+
+                <div class="ls_message_body">
+                  <p>Start a [90] days free trial with our recommanded model, <a href="https://www.ibm.com/products/watsonx-code-assistant">IBM watsonx Code Assistant</a>.
```
Comment (manstis) [inline ansible_ai_connect/users/templates/users/trial.html:63]: Typo: "recommended" vs "recommanded"

File: ansible_ai_connect/users/templates/users/trial/terms.html [hunk 4]
```diff
@@ -0,0 +1,361 @@
...
+                            Click to accept both the Red Hat Ansible Lightspeed
+                            Technical Preview terms and the IBM watsonx Code Assistant
+                            Technical Preview terms.
+                        </p>
+                        <div class="ls_terms_accept_buttons"></div>
+                        <button class="pf-c-button pf-m-primary" type="submit" name="accept_trial_terms" value="True">
+                            <span class="pf-c-button__icon"><i class="fa fa-check-circle"></i></span>
+                            Agree
+                        </button>
+                        <button class="pf-c-button pf-m-secondary" type="submit" name="accepted" value="False">
```
Comment (manstis) [inline ansible_ai_connect/users/templates/users/trial/terms.html:351]: Should the `name` be `accept_trial_terms`?

File: ansible_ai_connect/users/templates/users/trial/terms.html
Comment (manstis) [inline ansible_ai_connect/users/templates/users/trial/terms.html:1]: Presumably the wording is under review?

**Review by manstis — APPROVED** [review]
Minor comments; otherwise LGTM :+1:

**Comment by manstis** [comment]
You possibly want to check the footer (horizontal panel containing the buttons/links).

The one for "One Click Trial" is taking up the whole width of the block above; whereas our other "Home" pages are different and the applicable HTML block is centred and the buttons/links spaced better.

image

**Comment by goneri** [comment]
> You possibly want to check the footer (horizontal panel containing the buttons/links).
> 
> The one for "One Click Trial" is taking up the whole width of the block above; whereas our other "Home" pages are different and the applicable HTML block is centred and the buttons/links spaced better.

Nice catch. This is caused by the wider string that enlarge the block width. We have a meeting later with Ran to adjust this kind of details. It could also be nice to move this barre in a different template fragment, to avoid code duplication.

**Pull request merged**